core/src/main/java/uk/gov/justice/services/core/jms/JmsSender.java

```diff
@@ -74,16 +74,13 @@ private void send(final String queueName, final Envelope envelope) {
         try {
             final Queue queue = (Queue) getInitialContext().lookup(queueName);
 
-            try (QueueConnection queueConnection = queueConnectionFactory.createQueueConnection()) {
+            try (QueueConnection queueConnection = queueConnectionFactory.createQueueConnection();
+                 QueueSession session = queueConnection.createQueueSession(false, QueueSession.AUTO_ACKNOWLEDGE);
+                 QueueSender sender = session.createSender(queue)) {
 
-                try (QueueSession session = queueConnection.createQueueSession(false, QueueSession.AUTO_ACKNOWLEDGE)) {
+                sender.send(envelopeConverter.toMessage(envelope, session));
```

Comment:
Yes, all the resources will be closed in reverse order of the order in which they are created / listed inside the parentheses. http://tutorials.jenkov.com/java-exception-handling/try-with-resources.html#using-multiple-resources